ragna/source_storages/_lancedb.py

```diff
@@ -78,9 +101,13 @@ def store(
                 table.add(
                     [
                         {
+                            field: "''" for field in all_fields
+                        }  # ensure all columns exist
+                        | {
                             "id": str(uuid.uuid4()),
                             "document_id": str(document.id),
                             "document_name": str(document.name),
+                            **document.metadata,
```

Comment:
Cont'd from suggestion above

```suggestion
                            # Unpacking the default metadata first
                            # so it can be overridden by the concrete
                            # values if available
                            **extra_metadata
                            **document.metadata,
```